crates/valence_core/src/text/text_format.rs

```diff
@@ -0,0 +1,418 @@
+//! Provides the [`TextFormat`] trait and implementations.
+
+use std::borrow::Cow;
+
+use super::{ClickEvent, Color, Font, HoverEvent, IntoText, Text};
+
+/// Provides the methods necessary for working with [`Text`] objects.
+///
+/// This trait exists to allow using [`IntoText`] types without
+/// having to first convert the type into [`Text`].
+///
+/// # Usage
+///
+/// ```
+/// # use valence_core::text::{Text, color::NormalColor, TextFormat};
+/// # fn usage_example(my_text: &mut Text) {
+/// my_text.color(NormalColor::Red).bold();
+/// my_text.add_child("CRABBBBB".obfuscated());
+/// # }
+pub trait TextFormat {
```

Comment:
Now that we have `IntoText`, I think we should just merge `TextFormat` with `IntoText`. To reduce redundancy, I am also in favor of removing `ReturnType` since the fields of `TextInner` are publicly accessible now, but I'm open to other ideas.